src/test/rustdoc/issue-55364.rs

```diff
@@ -0,0 +1,85 @@
+// ignore-tidy-linelength
+
+// First check the four top level modules are present
+// @has issue_55364/index.html
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="mod"][@href="subone/index.html"]' 'subone'
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="mod"][@href="subtwo/index.html"]' 'subtwo'
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="mod"][@href="subthree/index.html"]' 'subthree'
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="mod"][@href="subfour/index.html"]' 'subfour'
+
+// Now a module with inner documentation
+
+// @has issue_55364/subone/index.html
+// These foo/bar links in the module's documentation should refer inside `subone`
+// @has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/subone/fn.foo.html"]' 'foo'
+// @has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/subone/fn.bar.html"]' 'bar'
+pub mod subone {
+    //! See either [foo] or [bar].
+
+    /// See [bar]
+    pub fn foo() {}
+    /// See [foo]
+    pub fn bar() {}
+}
+
+// A module with outer documentation
+
+// @has issue_55364/subtwo/index.html
+// These foo/bar links in the module's documentation should not reference inside `subtwo`
+// @!has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/subtwo/fn.foo.html"]' 'foo'
+// @!has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/subtwo/fn.bar.html"]' 'bar'
+// Instead it should be referencing the top level functions
+// @has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/fn.foo.html"]' 'foo'
+// @has - '//section[@id="main"]/div[@class="docblock"]//a[@href="../../issue_55364/fn.bar.html"]' 'bar'
+// Though there should be such links later
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="fn"][@href="fn.foo.html"]' 'foo'
+// @has - '//section[@id="main"]/table//tr[@class="module-item"]/td/a[@class="fn"][@href="fn.bar.html"]' 'bar'
+/// See either [foo] or [bar].
+pub mod subtwo {
+
+    /// See [bar]
+    pub fn foo() {}
```

Comment:
Can you check for those two functions too please?

Reply:
I've added checks that these two refer to one another inside `subtwo` despite the module itself referring to the outer foo/bar.